### PR TITLE
[ticket/14443] Introduce additional parameter to messenger

### PR DESCRIPTION
**Pull request by gn36**
See https://tracker.phpbb.com/browse/PHPBB3-14443

PHPBB3-14443

### CODE REVIEW

**Comment by Elsensee** [comment]
I have checked and phpBB itself doesn't use the `$template_path` parameter and to be honest, I doubt that any extension will use it. I guess we keep it anyway to keep some sort of BC? Somebody agreeing/disagreeing here?

**Comment by marc1706** [comment]
It's called a prefix but to me it seems like it's a suffix?

**Comment by gn36** [comment]
I actually thought the same thing. It is a suffix to the path, but a prefix to the template name, which is probably the reason it was named this way everywhere it is used originally (note that the original use was actually prefixing this path to the template name). The only reason I kept the name was to keep it the same as in the methods that call this one, so it is easier to track the function of the variable.

I can rename it if you want. Since this parameter is handed down from other methods, I would then suggest renaming them all, though.

File: phpBB/includes/functions_messenger.php [hunk 1]
```diff
@@ -232,7 +232,7 @@ function template($template_file, $template_lang = '', $template_path = '')
 		if ($template_path)
 		{
 			$template_paths = array(
-				$template_path,
+				$template_path . '/' . $template_dir_prefix,
```
Comment (Nicofuma) [inline phpBB/includes/functions_messenger.php]: Shouldn't it be a ternary to avoid an extra `/` if `$template_dir_prefix` is empty?

Comment (marc1706) [inline phpBB/includes/functions_messenger.php]: Yes, should only be added if the prefix is not empty.

**Comment by marc1706** [comment]
Ah ok, that's the thinking behind calling it a prefix. You can leave it at that, then.

File: phpBB/includes/functions_messenger.php (hunk 1 above)
Comment (gn36) [inline phpBB/includes/functions_messenger.php]: Oops, I will take care of that.

**Comment by Elsensee** [comment]
:+1:

File: phpBB/includes/functions_messenger.php [hunk 2]
```diff
@@ -210,10 +210,12 @@ function set_mail_priority($priority = MAIL_NORMAL_PRIORITY)
 	/**
 	* Set email template to use
 	*/
-	function template($template_file, $template_lang = '', $template_path = '')
+	function template($template_file, $template_lang = '', $template_path = '', $template_dir_prefix = '')
 	{
 		global $config, $phpbb_root_path, $phpEx, $user, $phpbb_extension_manager;
 
+		$template_dir_prefix = (!$template_dir_prefix || substr($template_dir_prefix, 0, 1) === '/') ? $template_dir_prefix : '/' . $template_dir_prefix;
```
Comment (nickvergessen) [inline phpBB/includes/functions_messenger.php]: `substr($template_dir_prefix, 0, 1)` is equivalent to `$template_dir_prefix[0]`, which should be faster, but in this case it is negligible in this case I guess

Comment (gn36) [inline phpBB/includes/functions_messenger.php]: @nickvergessen: I can certainly change that if you want me to.

Comment (marc1706) [inline phpBB/includes/functions_messenger.php]: I'd prefer that change, too. Please modify it when you have some time and we'll review the PR again once you've updated it.

Comment (gn36) [inline phpBB/includes/functions_messenger.php]: @marc1706: Done.

**Pull request merged**